gossip/apply_genesis.go

```diff
@@ -60,14 +60,16 @@ func (s *Store) ApplyGenesis(g genesis.Genesis) (err error) {
 	rules := s.GetRules()
 	gasLimit := rules.Blocks.MaxBlockGas
 
-	s.SetBlock(0, inter.NewBlockBuilder().
+	blockZero := inter.NewBlockBuilder().
 		WithNumber(0).
-		WithTime(evmcore.FakeGenesisTime-1). // TODO: extend genesis generator to provide time
+		WithTime(evmcore.FakeGenesisTime - 1).
```

Comment:
It would be better to add some way to extract this information from `genesis.Genesis` than to use a non-constant value.